File: src/main/js/libgdx/gdx/utils/viewport/FillViewport.js

import Scaling from 'gdx/utils/Scaling'
import ScalingViewport from 'gdx/utils/viewport/ScalingViewport'

/**
 * @JSName("gdx.utils.viewport.FillViewport")
 */
export default class FillViewport extends ScalingViewport {
    constructor(worldWidth, worldHeight, camera) {
        super(Scaling.fill, worldWidth, worldHeight, camera)
    }
}

File: src/main/scala/com/badlogic/gdx/utils/viewport/FillViewport.scala
package com.badlogic.gdx.utils.viewport

import scala.scalajs.js
import scala.scalajs.js.annotation.JSName
import com.badlogic.gdx.graphics._

/** A ScalingViewport that uses {@link Scaling#fill} so it keeps the aspect ratio by scaling the world up to take the whole screen
  * (some of the world may be off screen).
  * @author Daniel Holderbaum
  * @author Nathan Sweet */
@JSName("gdx.utils.viewport.FillViewport")
@js.native
class FillViewport extends ScalingViewport {
  /** Creates a new viewport using a new {@link OrthographicCamera}. */
  def this(worldWidth: Float, worldHeight: Float, camera: Camera = ???) = this()
}

